apps/src/music/blockly/blockUtils.js

```diff
@@ -143,3 +143,34 @@ function simple2FunctionCallGenerator(functionName) {
     Sequencer.endFunctionContext();
   `;
 }
+
+// For a given block id, return a list of block types. These block types
+// represent any C-shaped block between itself and the root (top) block
+// which contains it. The returned list could include types for loop blocks,
+// function definitions, conditionals, or other control structures.
+// These blocks all have a "statement" input that contains other blocks.
+export function findParentStatementInputTypes(id) {
+  if (id === 'preview') {
+    return [];
+  }
+
+  // Ensure Blockly is defined for the sake of unit tests.
+  const block = global.Blockly && Blockly.getMainWorkspace().getBlockById(id);
```

Comment:
This is a hack and I don't love it. Without the `global.Blockly` check, unit tests begin failing. Example:

```

FAIL test/unit/music/player/sequencer/Simple2SequencerTest.ts (6.863 s)
--
1201 | ● Simple2Sequencer › starts a new sequence at the given measure
1202 |  
1203 | ReferenceError: Blockly is not defined
1204 |  
1205 | 154 \|     return [];
1206 | 155 \|   }
1207 | > 156 \|   const block = Blockly.getMainWorkspace().getBlockById(id);
1208 | \|                 ^
1209 | 157 \|   const parentTypes = [];
1210 | 158 \|   function addParentBlockTypes(currentBlock) {
1211 | 159 \|     if (currentBlock) {
1212 |  
1213 | at Blockly (src/music/blockly/blockUtils.js:156:17)
1214 | at Simple2Sequencer.playSound (src/music/player/sequencer/Simple2Sequencer.ts:225:56)
1215 | at Object.<anonymous> (test/unit/music/player/sequencer/Simple2SequencerTest.ts:46:15)
```

Reply:
Hmm, don't we have some `setupBlocklyGlobal` test hook that we sometimes include in front end tests to get around this? FWIW that also does feel a bit hacky to me, but maybe we need to do something similar in music tests to avoid this check?

Maybe with google blockly, it's as simple as doing `window.Blockly = initializeGoogleBlocklyWrapper(GoogleBlockly);` as part of the unit test (though probably in a separate hook still, not directly in the unit test file)?

Reply:
I was able to clean this up by creating a `setupGoogleBlockly` test hook!

https://github.com/code-dot-org/code-dot-org/pull/61997/commits/b417261ac5a607a57bdb3ce4cea7c9215baa174e